src/drivers/distance_sensor/cm8jl65/cm8jl65.cpp

```diff
@@ -0,0 +1,784 @@
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file cm8jl65.cpp
+ * @author Claudio Micheli <claudio@auterion.com>
+ *
+ * Driver for the Lanbao PSK-CM8JL65-CC5 distance sensor.
+ * Make sure to disable MAVLINK messages (MAV_0_CONFIG PARAMETER)
+ * on the serial port you connect the sensor,i.e TELEM1.
+ *
+ */
+
+#include <px4_config.h>
+#include <px4_getopt.h>
+#include <px4_workqueue.h>
+
+#include <sys/types.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include <poll.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <termios.h>
+
+#include <perf/perf_counter.h>
+
+#include <drivers/drv_hrt.h>
+#include <drivers/drv_range_finder.h>
+#include <drivers/device/device.h>
+#include <drivers/device/ringbuffer.h>
+
+#include <uORB/uORB.h>
+#include <uORB/topics/distance_sensor.h>
+
+#include "cm8jl65_parser.h"
+
+/* Configuration Constants */
+
+#ifndef CONFIG_SCHED_WORKQUEUE
+# error This requires CONFIG_SCHED_WORKQUEUE.
+#endif
+
+#define CM8JL65_TAKE_RANGE_REG		'd'
+
+// designated serial port on Pixhawk (TELEM2)
+#define CM8JL65_DEFAULT_PORT		"/dev/ttyS2" // Its baudrate is 115200
+
+// normal conversion wait time
+#define CM8JL65_CONVERSION_INTERVAL 50*1000UL/* 50ms */
+
+
+class CM8JL65 : public cdev::CDev
+{
+public:
+
+	// Constructor
+	CM8JL65(const char *port = CM8JL65_DEFAULT_PORT, uint8_t rotation = distance_sensor_s::ROTATION_DOWNWARD_FACING);
+
+	// Virtual destructor
+	virtual ~CM8JL65();
+
+	virtual int  init();
+	virtual int  ioctl(device::file_t *filp, int cmd, unsigned long arg);
+
+	/**
+	* Diagnostics - print some basic information about the driver.
+	*/
+	void				print_info();
+
+private:
+
+	char 				             _port[20];
+	uint8_t                   _rotation;
+	float				             _min_distance;
+	float				             _max_distance;
+	int         	             _conversion_interval;
+	work_s				             _work{};
+	ringbuffer::RingBuffer	  *_reports;
+	int				               _fd;
+	uint8_t			             _linebuf[25];
+	uint8_t                   _cycle_counter;
+
+	enum CM8JL65_PARSE_STATE	 _parse_state;
+	unsigned char             _frame_data[4];
```

Comment:
Can you use a define for the length here and use it for the parsing as well? `int cm8jl65_parser(uint8_t c, uint8_t parserbuf[PARSER_BUF_LENGTH], CM8JL65_PARSE_STATE *state, uint16_t *crc16, int *dist)`